carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp

```diff
@@ -14,108 +14,151 @@
-                cav_msgs::MobilityOperation traffic_mobility_msg = traffic_worker_.mobilityMessageGenerator(traffic_worker_.getPinPoint());      
-                
-                //start constantly broadcasting mobilityOperation msg
-                traffic_mobility_operation_pub_.publish(traffic_mobility_msg);
-              }
-            return true;
-        });
-    ROS_INFO_STREAM("Traffic Incident node is initialized...");
-    ros::CARMANodeHandle::spin();
+#include "traffic_incident_node.hpp"
+
+namespace traffic
+{
+namespace std_ph = std::placeholders;
+
+TrafficIncidentNode::TrafficIncidentNode(const rclcpp::NodeOptions & options)
+: carma_ros2_utils::CarmaLifecycleNode(options),
+  traffic_worker_(
+    std::bind(&TrafficIncidentNode::publishTrafficIncidentMobilityOperation, this, std_ph::_1))
+{
+  declare_parameter<std::string>("sender_id", sender_id_);
+  declare_parameter<std::string>("event_reason", event_reason_);
+  declare_parameter<std::string>("event_type", event_type_);
+  declare_parameter<double>("down_track", down_track_);
+  declare_parameter<double>("up_track", up_track_);
+  declare_parameter<double>("min_gap", min_gap_);
+}
+
+carma_ros2_utils::CallbackReturn TrafficIncidentNode::handle_on_configure(
+  const rclcpp_lifecycle::State &)
+{
+  get_parameter<std::string>("sender_id", sender_id_);
+  get_parameter<std::string>("event_reason", event_reason_);
+  get_parameter<std::string>("event_type", event_type_);
+  get_parameter<double>("down_track", down_track_);
+  get_parameter<double>("up_track", up_track_);
+  get_parameter<double>("min_gap", min_gap_);
+
+  traffic_worker_.setSenderId(sender_id_);
+  traffic_worker_.setDownTrack(down_track_);
+  traffic_worker_.setUpTrack(up_track_);
+  traffic_worker_.setMinGap(min_gap_);
+  traffic_worker_.setEventReason(event_reason_);
+  traffic_worker_.setEventType(event_type_);
+
+  // Setup pub/sub
+  pinpoint_driver_sub_ = create_subscription<gps_msgs::msg::GPSFix>(
+    "gps_common_fix",
+    10,
+    std::bind(&TrafficIncidentWorker::pinpointDriverCallback, &traffic_worker_, std_ph::_1));
+  traffic_mobility_operation_pub_ =
+    create_publisher<carma_v2x_msgs::msg::MobilityOperation>("outgoing_mobility_operation", 10);
+
+  // setup services
+  start_broadcast_request_service_server = create_service<carma_msgs::srv::SetTrafficEvent>(
+    "start_broadcasting_traffic_event",
+    std::bind(
+      &TrafficIncidentNode::startTrafficBroadcastCallback,
+      this,
+      std_ph::_1,
+      std_ph::_2,
+      std_ph::_3));
+  stop_broadcast_request_service_server = create_service<std_srvs::srv::Trigger>(
+    "stop_broadcasting_traffic_event",
+    std::bind(
+      &TrafficIncidentNode::stopTrafficBroadcastCallback,
+      this,
+      std_ph::_1,
+      std_ph::_2,
+      std_ph::_3));
+
+  // spin loop
+  spin_timer_ = create_timer(
+    get_clock(),
+    std::chrono::milliseconds(100),
+    std::bind(&TrafficIncidentNode::spin_callback, this));
```

Comment:
let's move the timer to handle_on_activate function so the timer starts only when the node is activated.